### PR TITLE
feat: past clients

### PR DESCRIPTION
**Pull request by theogainey**
### Description
add PastClients components
- PastClients will only appear if there are past clients

### Spec

Designs: [Designs](https://www.figma.com/file/kswpPSHH2h5njVyfGDbIHk/FSA-Capstone-Project-2022?node-id=0%3A1)

See Story: [199](https://sparkbox.atlassian.net/browse/FSA22V1-199)

### Validation

* [x] This PR has visual elements, so it was reviewed by a designer.
* [x] This PR has code changes, and our linters still pass.
* [x] This PR affects production code, so it was browser tested (see below).
* [x] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
2. Pull down all related branches.
3. Confirm all tests pass.

### CODE REVIEW

File: components/PastClients.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default function PastClients({ pastClients }) {
+  return (
+    <section aria-labelledby="Skills-Header" className="cmp-past-clients">
+      <h2 id="Skills-Header" className="cmp-past-clients__heading">Past Clients</h2>
```
Comment (QuinnrElder) [inline components/PastClients.jsx:4]: Lets update these ID's to be uniqu
Screen Shot 2022-05-20 at 8 49 33 AM
e

File: styles/components/_past-clients.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  &__client {
+    border: 1px color("white") solid;
+    display: inline;
+    font-size: font-size("md");
+    font-weight: 300;
+    margin-right: space("md");
+    padding: space("xs");
+    text-transform: capitalize;
+  }
```
Comment (QuinnrElder) [inline styles/components/_past-clients.scss:24]: Love how things are alphabatized!

File: components/PastClients.jsx (hunk 1 above)
Comment (theogainey) [reply components/PastClients.jsx]: When copy and past goes wrong

**Review by QuinnrElder — DISMISSED** [review]
Everything looks good here to me! I would just make sure a designer looks over things and we should be good to go. Good job!

File: components/PastClients.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export default function PastClients({ pastClients }) {
```
Comment (arnest00) [inline components/PastClients.jsx:1]: Super clean component

File: styles/components/_skills-grid.scss [hunk 4]
```diff
@@ -1,7 +1,5 @@
 .cmp-skills-grid {
   width: 100%;
-  margin: 0 auto;
-  padding: 0 space("lg");
```
Comment (arnest00) [inline styles/components/_skills-grid.scss:4]: Oh, cool. I also tried to take care of this in my current card, hah.

**Review by arnest00 — CHANGES_REQUESTED** [review]
The Past Clients section looks great! One thing to note: All tests are passing, but we do get this warning.

Screen Shot 2022-05-20 at 11 25 35 AM

Adding a weeklyCapacity and a forecastedHours to our `MOCK_TEAM_MEMBER` in `[id].test.jsx`, and as well as the assertion on line 45 in that file should take care of that, though!

**Comment by QuinnrElder** [comment]
>

**Comment by QuinnrElder** [comment]
My bad! Was trying to say nice catch to Osmond but closed the PR ~ sorry!

**Comment by theogainey** [comment]
Unless someone has a very good reason I am going to delete the tests in `[id].test.jsx ` cause these tests are now redundant. For the first test, everything related to how `[id].jsx` handles a name that will cause problems if it breaks is handled in `utils/getFullName.jsx` and is tested in `utils/getFullName.test.jsx `. The same idea goes for the second one. How the data is passed to the page is handled in `services/apiService.js` and is tested in `services/apiService.test.js`. Keeping these tests we are going to keep getting errors when things don't break and the only time we will get an error, that isn't also caught in another test, that tells us something broke would be if NextJS is the thing that is broke

**Comment by QuinnrElder** [comment]
IMO ~ I wouldn't delete the tests for ```[id].test.jsx``` just yet. Yes, the test that is currently #1 in that file is a redundant test. But this test can be refactored into an integration test. We are rendering ```<DetailPage fetchedTeamMember={MOCK_TEAM_MEMBER} />``` So we could not only test that a name rendered (current test) but we could also test all the sub-components in ```<DetailPage/>```rendered correctly. This is where we could also test functionality like clicking links and dom changes when interacting with the page.

It's not too uncommon to see tests that are very similar in nature across multiple testing files. The difference her is that we can test child components through their parent component.

I believe that it would be the same for the async tests...

Thoughts on this?
We can hop on a call to talk things through and make sure that we are all on the same page.

**Comment by theogainey** [comment]
If we are going to do integration tests I believe that would we need to add something more than jest like Cypress and I am also am under the understanding that integration tests are currently beyond the scope of what we are currently trying to achieve but I will have to defer to @dustin-jw on that

**Comment by dustin-jw** [comment]
> If we are going to do integration tests I believe that would we need to add something more than jest like Cypress and I am also am under the understanding that integration tests are currently beyond the scope of what we are currently trying to achieve but I will have to defer to @dustin-jw on that

I think we ought to be able to stick with Jest for integration tests, but we'd want them to be more about the general structure than component specifics. Like we could render the Details page and make sure that the heading from each section that is supposed be on the page is there – something simple like that would be fine for now, and when the whole page is built out, we could come back to it make it a little more robust.

**Comment by QuinnrElder** [comment]
Very true @theogainey ~ Cypress is a great tool for integration testing!! It can be done in jest it is just not fun ~ it's not automated and much more granular of an approach. I apologize too ~ I was not aware that integration testing was not included in the current scope.

**Comment by theogainey** [comment]
Test thing taken care of and Design is what it should be

**Review by arnest00 — APPROVED** [review]

**Review by QuinnrElder — APPROVED** [review]
Nice teamwork on this PR!

**Pull request merged**